brainiak/reprsimil/brsa.py

```diff
@@ -87,106 +227,220 @@ class BRSA(BaseEstimator):
-        imposed on the log(SNR), this range should not be too
-        large. 10 is a pretty loose range. This parameter is
-        used in a half-Cauchy prior on the standard deviation
-    init_iter: how many initial iterations to fit the model
+    tau_range: float. Default: 5.0
+        The reasonable range of the standard deviation
+        of log(SNR). This range should not be too
+        large. 5 is a loose range.
+        When a Gaussian Process is imposed on the log(SNR),
+        this parameter is used in a half-Cauchy prior
+        on the standard deviation, or an inverse-Gamma prior
+        on the variance of the GP.
+    tau2_prior: callable[[float, int, float], [float, float]],
+        Default: prior_GP_var_inv_gamma.
+        Can be prior_GP_var_inv_gamma or prior_GP_var_half_cauchy,
+        or a custom function.
+        The function which impose a prior for tau^2, the variance of the
+        GP prior on log(SNR), and returns the MAP estimate of tau^2.
+        It can be either prior_GP_var_inv_gamma for inverse-Gamma
+        or prior_GP_var_half_cauchy for half-Cauchy.
+        half-Cauchy prior is in fact imposed on tau.
+        But tau_range describes the range of tau in the prior in both cases.
+        Both functions are part of brsa module.
+        See also `.prior_GP_var_inv_gamma` and
+        `.prior_GP_var_half_cauchy`
+        To use the default inverse-Gamma prior, you can ignore this argument::
+
+            from brainiak.reprsimil.brsa import BRSA
+            brsa = BRSA()
+
+        If you want to try the alternative half-Cauchy prior,
+        then you need to import it in addition to BRSA::
+
+            from brainiak.reprsimil.brsa import BRSA, prior_GP_var_half_cauchy
+            brsa = BRSA(tau2_prior=prior_GP_var_half_cauchy)
+
+    eta: float. Default: 0.0001
+        A small number added to the diagonal element of the
+        covariance matrix in the Gaussian Process prior. This is
+        to ensure that the matrix is invertible.
+    init_iter: int. Default: 20
+        How many initial iterations to fit the model
         without introducing the GP prior before fitting with it,
         if GP_space or GP_inten is requested. This initial
         fitting is to give the parameters a good starting point.
-    optimizer: the optimizer to use for minimizing cost function.
-        We use 'BFGS' as a default. Users can try other optimizer
-        coming with scipy.optimize.minimize, or a custom
-        optimizer.
-    rand_seed : int, default: 0
+    optimizer: str or callable. Default: 'BFGS'
+        The optimizer to use for minimizing cost function.
+        We use 'BFGS' as a default. Users can try other strings
+        corresponding to optimizer provided by scipy.optimize.minimize,
+        or a custom optimizer.
+    rand_seed : int. Default: 0
         Seed for initializing the random number generator.
+    anneal_speed: float. Default: 20
```

Comment:
This needs further test to decide whether annealing is necessary. I prefer deferring this test for now.